lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivationSpec.hs

```diff
@@ -0,0 +1,315 @@
+                \15, 18 or 21 words are expected.")
+
+        it "early error reported first (Wrong number of words - 2)" $ do
+            let res = fromMnemonic @'[15] @"testing"
+                        ["mom", "unveil", "slim", "abandon"
+                        , "nut", "cash", "laugh", "impact"
+                        , "system", "split", "depth", "sun"
+                        ]
+            res `shouldBe` Left (FromMnemonicError "Invalid number of words: \
+                \15 words are expected.")
+
+        it "early error reported first (Error not in first constructor)" $ do
+            let res = fromMnemonic @'[15,18,21,24] @"testing"
+                        ["盗", "精", "序", "郎", "赋", "姿", "委", "善", "酵"
+                        ,"祥", "赛", "矩", "蜡", "注", "韦", "效", "义", "冻"
+                        ]
+            res `shouldBe` Left (FromMnemonicError "Found invalid (non-English) \
+                \word: \"盗\".")
+
+        it "early error reported first (Error not in first constructor)" $ do
+            let res = fromMnemonic @'[12,15,18] @"testing"
+                        ["盗", "精", "序", "郎", "赋", "姿", "委", "善", "酵"
+                        ,"祥", "赛", "矩", "蜡", "注", "韦", "效", "义", "冻"
+                        ]
+            res `shouldBe` Left (FromMnemonicError "Found invalid (non-English) \
+                \word: \"盗\".")
+
+        it "successfully parse 15 words in [15,18,21]" $ do
+            let res = fromMnemonic @'[15,18,21] @"testing"
+                        ["cushion", "anxiety", "oval", "village", "choose"
+                        , "shoot", "over", "behave", "category", "cruise"
+                        , "track", "either", "maid", "organ", "sock"
+                        ]
+            res `shouldSatisfy` isRight
+
+        it "successfully parse 15 words in [12,15,18]" $ do
+            let res = fromMnemonic @'[12,15,18] @"testing"
+                        ["cushion", "anxiety", "oval", "village", "choose"
+                        , "shoot", "over", "behave", "category", "cruise"
+                        , "track", "either", "maid", "organ", "sock"
+                        ]
+            res `shouldSatisfy` isRight
+
+        it "successfully parse 15 words in [9,12,15]" $ do
+            let res = fromMnemonic @'[9,12,15] @"testing"
+                        ["cushion", "anxiety", "oval", "village", "choose"
+                        , "shoot", "over", "behave", "category", "cruise"
+                        , "track", "either", "maid", "organ", "sock"
+                        ]
+            res `shouldSatisfy` isRight
+
+    describe "Keys storing and retrieving roundtrips" $ do
+        it "XPriv" (property prop_roundtripXPriv)
+        it "XPub" (property prop_roundtripXPub)
+
+
+{-------------------------------------------------------------------------------
+                               Properties
+-------------------------------------------------------------------------------}
+
+
```

Comment:
And here.

https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#use-only-a-single-blank-line-between-top-level-definitions

Reply:
Thanks, fixed.